### PR TITLE
Fix async issue in HOC wrapping getServerSideProps

### PR DESCRIPTION
**Pull request by 39otsu**
## Title
Fix async issue in HOC wrapping getServerSideProps

## Issue Number
Issue #211 

## Description
This pull request addresses the bug where the high order component (HOC) wrapping around `getServerSideProps` to protect routes was not returning properly due to an asynchronous function call in a non-async component. The issue stemmed from the return statement awaiting a function passed through the parameters, which was not possible in a non-async component. To fix this, I made the component async and properly awaited the function, ensuring that the HOC returns the result of `getServerSideProps` correctly.

## Testing
I have verified that this does not mess up the tests.

## Possible Errors
- Edge cases where the authentication criteria are met but the route is still inaccessible.
- Unexpected interactions with other components or middleware in the project.

## Images
N/A

Credit(s):
MacGyver Codilla
Justin Vance Llamas

### CODE REVIEW

**Review by jusasin — APPROVED** [review]
Great work on finding the issue that was causing the problems. I know it was holding us back from completing other task. Good job.

File: small-talk/components/smolAuth.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ export function smolAuth(getServerSidePropsFunc) {
                 let isFound = (json.found != null) ? json.found : false;
 
                 if (isFound == false) {
+                    //console.log("SMOLAUTH: NOT FOUND")
```
Comment (jusasin) [inline small-talk/components/smolAuth.js:33]: love the use of console logs to figure out what is wrong

**Comment by aiharc1** [comment]
I'm unsure if this is related directly to this PR or not but I'm not sure if the protected routes are working properly. I can manipulate the url and go '/homepage' while not being logged in. Was that what you meant when you mentioned the wrapping to protect routes ?

**Review by aiharc1 — APPROVED** [review]
Looks good. I saw no issues besides the possibility of url manipulation. Unsure if that is part of scope

**Review by AlvaroH2001 — APPROVED** [review]
Great job on fixing this bug, the changes were minimal but the effects profound.
